x-pack/legacy/plugins/monitoring/public/components/cluster/overview/logstash_panel.js

```diff
@@ -21,19 +21,58 @@ import {
   EuiDescriptionListDescription,
   EuiHorizontalRule,
   EuiIconTip,
+  EuiToolTip,
+  EuiBadge
 } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
 import { i18n } from '@kbn/i18n';
+import { get } from 'lodash';
 
 export function LogstashPanel(props) {
-  if (!props.node_count) {
+  const { setupMode } = props;
+  const nodesCount = props.node_count || 0;
+  const queueTypes = props.queue_types || {};
+
+  // Do not show if we are not in setup mode
+  if (!nodesCount && !setupMode.enabled) {
```

Comment:
Should this maybe be `if (!nodesCount || !setupMode.enabled) {` so if only one of the conditions is false return null

Reply:
It needs to be both. If setup mode is enabled, we want to always show the panel, even if there are no nodes.